scripts/menu/server/sv_menu.lua

```diff
@@ -0,0 +1,405 @@
+  local src = source
+  if type(x) ~= 'number' or type(y) ~= 'number' or type(z) ~= 'number' then return end
+    
+  local allow = PlayerHasTxPermission(src, 'players.teleport')
+  TriggerEvent("txaLogger:menuEvent", src, "teleportCoords", true, { x = x, y = y, z = z })
+  if allow then
+    TriggerClientEvent('txAdmin:menu:tpToCoords', src, x, y, z)
+  end
+end)
+
+RegisterServerEvent('txAdmin:menu:tpToPlayer', function(id)
+  local src = source
+  local parseId = type(id) == 'string' and tonumber(id) or id
+  if not parseId then
+    return
+  end
+
+  local allow = PlayerHasTxPermission(src, 'players.teleport')
+  local data = { x = nil, y = nil, z = nil, playerName = nil }
+  data.playerName = GetPlayerName(parseId)
+  if allow then
+    -- ensure the player ped exists
+    local ped = GetPlayerPed(parseId)
+    if ped then
+      local coords = GetEntityCoords(ped)
+      data.x = coords[1]
+      data.y = coords[2]
+      data.z = coords[3]
+      TriggerClientEvent('txAdmin:menu:tpToCoords', src, data.x, data.y, data.z)
+    end
+  end
+  TriggerEvent('txaLogger:menuEvent', src, 'teleportPlayer', allow, data)
+end)
+
+RegisterServerEvent('txAdmin:menu:summonPlayer', function(id)
+  local src = source
+  local parseId = type(id) == 'string' and tonumber(id) or id
+  if not parseId then
+    return
+  end
+  local allow = PlayerHasTxPermission(src, 'players.teleport')
+  if allow then
+    -- ensure the player ped exists
+    local ped = GetPlayerPed(src)
+    if ped then
+      local coords = GetEntityCoords(ped)
+      TriggerClientEvent('txAdmin:menu:tpToCoords', parseId, coords[1], coords[2], coords[3])
+    end
+  end
+  local playerName = GetPlayerName(parseId)
+  TriggerEvent('txaLogger:menuEvent', src, 'summonPlayer', allow, playerName)
+end)
+
+RegisterServerEvent('txAdmin:menu:tpToWaypoint', function()
+  local src = source
+  local allow = PlayerHasTxPermission(src, 'players.teleport')
+  if allow then
+    TriggerClientEvent('txAdmin:menu:tpToWaypoint', src)
+    Wait(250)
+    local coords = GetEntityCoords(GetPlayerPed(src))
+    TriggerEvent("txaLogger:menuEvent", src, "teleportWaypoint", true,
+      { x = coords[1], y = coords[2], z = coords[3] })
+  else
+    TriggerEvent("txaLogger:menuEvent", src, "teleportWaypoint", false)
+  end
+end)
+
+RegisterServerEvent('txAdmin:menu:sendAnnouncement', function(message)
+  local src = source
+  local allow = PlayerHasTxPermission(src, 'players.message')
+  TriggerEvent("txaLogger:menuEvent", src, "announcement", allow, message)
+  if allow then
+    TriggerClientEvent('txAdmin:receiveAnnounce', -1, message)
+  end
+end)
+
+RegisterServerEvent('txAdmin:menu:fixVehicle', function()
+  local src = source
+  local allow = PlayerHasTxPermission(src, 'menu.vehicle')
+  TriggerEvent("txaLogger:menuEvent", src, "vehicleRepair", allow)
+  if allow then
+    TriggerClientEvent('txAdmin:menu:fixVehicle', src)
+  end
+end)
+
+local CREATE_AUTOMOBILE = GetHashKey("CREATE_AUTOMOBILE")
```

Comment:
`.

Also, see client comment: you should not use CREATE_AUTOMOBILE on an arbitrary model index as it may be another vehicle type leading to an inconsistent net clone.

Note, yet again, that CREATE_VEHICLE is a bit deferred and won't get a network ID until a client created it (i.e. DOES_ENTITY_EXIST loop on server).